### PR TITLE
Fix sending of cross shard scheduled transactions

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed sending of cross shard scheduled txs

### CODE REVIEW

File: process/block/preprocess/scheduledTxsExecution_test.go [hunk 1]
```diff
@@ -1653,3 +1653,5 @@ func TestScheduledTxsExecution_IsMiniBlockExecuted(t *testing.T) {
 	ok = scheduledTxsExec.IsScheduledTx(hash2)
 	assert.False(t, ok)
 }
+
+//TODO: Add unit tests for methods: GetScheduledTx
```
Comment (iulianpascalau) [inline process/block/preprocess/scheduledTxsExecution_test.go:1657]: TODOs

File: process/block/preprocess/transactions.go [hunk 2]
```diff
@@ -742,9 +742,43 @@ func (txs *transactions) CreateBlockStarted() {
...
+
+		for _, txHash := range mb.TxHashes {
+			tx, err := txs.scheduledTxsExecutionHandler.GetScheduledTx(txHash)
+			if err != nil {
+				log.Warn("transactions.addTxsFromScheduledMiniBlocks: GetScheduledTx", "tx hash", txHash, "error", err.Error())
+				continue
+			}
+
+			txs.txsForCurrBlock.mutTxsForBlock.Lock()
+			txs.txsForCurrBlock.txHashAndInfo[string(txHash)] = &txInfo{tx: tx, txShardInfo: txShardInfoToSet}
```
Comment (iulianpascalau) [inline process/block/preprocess/transactions.go:776]: so the fix was to add in the txsForCurrBlock struct the scheduled txs?

Comment (SebastianMarian) [reply process/block/preprocess/transactions.go:776]: Yep

File: process/block/preprocess/scheduledTxsExecution_test.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/preprocess/scheduledTxsExecution_test.go:1657]: Will do

**Review by raduchis — DISMISSED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**